src/phoenix/server/thread_server.py

```diff
@@ -34,9 +37,15 @@ def run_in_thread(self) -> Generator[Thread, None, None]:
         thread.start()
         time_limit = time() + 5  # 5 seconds
         try:
-            while not self.started and thread.is_alive() and time() < time_limit:
+            while (
+                time() < time_limit
+                and thread.is_alive()
+                and not self.should_exit
+                and not self.started
+            ):
                 sleep(1e-3)
-            if time() > time_limit:
+            if time() >= time_limit and not self.started:
+                self.should_exit = True
                 raise RuntimeError("server took too long to start")
```

Comment:
Does this while loop ever fail to satisfy the `not self.should_exit` condition? As soon as we set `self.should_exit` to `True`, a runtime error is raised. I am not seeing where `self.should_exit` is set initially.

Reply:
We don't know, but setting it to `True` is intended to shut it down regardless, because at that point we're going inform the user that it failed. It could continue to run, but we have no real control over it either way.